llap-server/src/java/org/apache/hadoop/hive/llap/io/api/impl/LlapInputFormat.java

```diff
@@ -138,10 +142,28 @@
       rr.start();
       return result;
     } catch (Exception ex) {
-      throw new IOException(ex);
+      Throwable rootCause = JavaUtils.findRootCause(ex);
```

Comment:
This needs to be added in HIveInputFormat as well? Can be a follow up ticket if the issue persists in tez (as in independent JVM case)

Reply:
good catch @rbalamohan , easily reproduced with tez driver, added a fix for that here:
before fix (task attempt failure): https://issues.apache.org/jira/secure/attachment/13037081/syslog
after fix (task attempt success): https://issues.apache.org/jira/secure/attachment/13037083/syslog_with_fix

please find fix in last commit, let me know if your +1 still holds, thanks!

Reply:
Thanks @abstractdog . LGTM. +1